models/iosxr/vrf/iosxr_vrf.yaml

```diff
@@ -0,0 +1,124 @@
+              description: VRF import
+              type: dict
+            advertise_as_vpn:
+              description: Advertise local EVPN imported routes to PEs
+              type: bool
+            default_vrf:
+              description: Export routes to default VRF
+              type: bool
+            vrf:
+              description: Export routes to a VRF
+              type: str
+            allow_imported_vpn:
+              description: Allow export of imported VPN routes to non-default VRF
+              type: bool
+            maximum:
+              description: Set maximum prefix limit
+              type: dict
+              suboptions:
+                prefix:
+                  description:  Set table's maximum prefix limit.
+                  type: int
+                threshold:
+                  description: mid-thresh (% of max).
+                  type: int
+        description:
+          description: Provides a short description of the VRF definition in the current active configuration.
+          type: str
+        evpn_route_sync:
+          description: EVPN Instance VPN ID used to synchronize the VRF route(s).
+          type: int
+        fallback_vrf:
+          description: Fallback VRF name
+          type: str
+        mhost:
+          description: Multicast host stack options
+          type: dict
+          suboptions:
+            afi:
+              description: Address Family Identifier (AFI)
+              type: str
+              choices: ['ipv4', 'ipv6']
+            default_interface:
+              description: Default interface for multicast.
+              type: str
+        rd:
+          description: VPN Route Distinguisher (RD).
+          type: str
+        remote_route_filtering:
+          description: Enable/Disable remote route filtering per VRF
+          type: bool
+        vpn:
+          description: VPN ID for the VRF
+          type: dict
+          suboptions:
+            id:
+              description: VPN ID for the VRF.
+              type: int
+  running_config:
+    description: The state the configuration should be left in.
+      - State I(purged) removes all the VRF configurations from the
+      target device. Use caution with this state.
+      - State I(deleted) only removes VRF attributes that this modules
+      manages and does not negate the VRF process completely. Thereby, preserving
+      address_family related configurations under VRF context.
+      - Running states I(deleted) and I(replaced) will result in an error if there
+      are address_family configuration lines present under a neighbor,
+      or a vrf context that is to be removed. Please use the
+      M(cisco.iosxr.iosxr_vrf) modules for prior cleanup.
+      - Refer to examples for more details.
+    type: str
+  state:
+    description: The state the configuration should be left in.
+    type: str
+    choices: [deleted, merged, replaced, gathered, rendered, parsed, purged]
```

Comment:
Not sure if `purged` is valid for this module. Can you share an example?

Reply:
- name: Purge Interface Configurations
  hosts: iosxr_devices
  gather_facts: no
  tasks:
    - name: Remove Interface Configurations
      iosxr_interface:
        name: GigabitEthernet1/0/1 
        state: purged
      delegate_to: localhost

Reply:
also found another example in the `cisco.ios vrf module docs`: https://docs.ansible.com/ansible/latest/collections/cisco/ios/ios_vrf_module.html#ansible-collections-cisco-ios-ios-vrf-module:~:text=%2D%20name%3A%20Configure%20set%20of%20VRFs%20and%20purge%20any%20others%0A%20%20cisco.ios.ios_vrf%3A%0A%20%20%20%20vrfs%3A%0A%20%20%20%20%20%20%2D%20red%0A%20%20%20%20%20%20%2D%20blue%0A%20%20%20%20%20%20%2D%20green%0A%20%20%20%20purge%3A%20true

Reply:
`state: purged` is only relevant for modules that have a parent-child relation ship with other modules. For example, `*_interfaces` (parent) -> `*_l2_interfaces`, `*_l3_interfaces`, `*_lag_interfaces`, etc.(child). In this case, running `state: deleted` on each of them individually would ONLY negate the attributes that they are responsible for handling. But not the interface itself (relevant for logical interfaces), i.e., before `state: purged` was introduced, there was no way to execute something like `no interface loopback0`.

The `state: purged` is like a "hard reset" where we bypass the rule of "a resource module should only affect attributes that it is responsible for", and cut off an entire "branch" of config. Same for the BGP resource modules. Running `state: deleted` on `*_bgp_global`, would NOT turn off the BGP process as a whole, instead it will only negate attributes from the BGP config section that the `*_bgp_global` module handles, leaving `*_address_family`, `*_neighbor_address_family`, etc., untouched. OTOH, `state: purged` executes `no bgp <ASN>` that removes the entire BGP config section and is equivalent to running `state: deleted` on all the BGP series modules.

Reply:
Got it!